src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1606,13 +1607,13 @@ internal StringBuilder AppendFormatHelper(IFormatProvider provider, String forma
                         if (startPos != pos)
                         {
                             // There was no brace escaping, extract the item format as a single string
-                            itemFormat = format.Substring(startPos, pos - startPos);
+                            itemFormatSpan = format.AsReadOnlySpan().Slice(startPos, pos - startPos);
```

Comment:
@justinvp, FYI this implements the optimization you suggested.

Reply:
Awesome! Thanks, @stephentoub!